docs/noencap-hybrid-modes.md

```diff
@@ -0,0 +1,174 @@
+# NoEncap and Hybrid Traffic Modes of Antrea
+
+Besides the default `Encap` mode, in which Pod traffic across Nodes will be
+encapsulated and sent over tunnels, Antrea also supports `NoEncap` and `Hybrid`
+traffic modes. In `NoEncap` mode, Antrea does not encapsulate Pod traffic, but
+relies on the Node network to route the traffic across Nodes. In `Hybrid` mode,
+Antrea encapsulates Pod traffic when the source Node and the destination Node
+are in different subnets, but does not encapsulate when the source and the
+destination Nodes are in the same subnet. This document describes how to
+configure Antrea with the `NoEncap` and `Hybrid` modes.
+
+## Hybrid Mode
+
+Let us start from `Hybrid` mode which is simpler to configure. `Hybrid` mode
+does not encapsulate Pod traffic when the source and the destination Nodes are
+in the same subnet. Thus it requires the Node network allow Pod IP addresses
+being sent out from the Nodes' NICs. This requirement is not supported in all
+the networks and clouds, or in some cases it might require specific
+configuration of the Node network. For example:
+
+* On AWS, the source/destination checks must be disabled on the EC2 instances of
+the Kubernetes Nodes, as described in the
+[AWS documentation](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_NAT_Instance.html#EIP_Disable_SrcDestCheck).
+
+* On Google Compute Engine, IP forwarding must be enabled on the VM instances as
+described in the [Google Cloud documentation](https://cloud.google.com/vpc/docs/using-routes#canipforward).
+
+* On Azure, there is no way to let VNet forward unknown IPs, hence Antrea
+`Hybrid` mode cannot work on Azure.
+
+If the Node network does allow Pod IPs sent out from the Nodes, you can
+configure Antrea to run in the `Hybrid` mode by setting the `trafficEncapMode`
+config option of `antrea-agent` to `hybrid`. The `trafficEncapMode` config
+option is defined in `antrea-agent.conf` of the `antrea` ConfigMap in the
+[Antrea deployment YAML](https://github.com/vmware-tanzu/antrea/blob/main/build/yamls/antrea.yml).
+
+```yaml
```

Comment:
maybe we should not include this snippet unless there is a strong reason to, as then we will need to keep it in sync I feel like the instructions above are clear enough.

Reply:
What you think about:
```
```yaml
trafficEncapMode: noEncap
...
```

Reply:
second one looks good to me